packages-exp/auth-exp/src/core/auth/initialize.ts

```diff
@@ -54,7 +55,12 @@ export function initializeAuth(app: FirebaseApp, deps?: Dependencies): Auth {
 
   if (provider.isInitialized()) {
     const auth = provider.getImmediate() as AuthImpl;
-    _fail(auth, AuthErrorCode.ALREADY_INITIALIZED);
+    const initialOptions = provider.getOptions() as Dependencies;
+    if (deepEqual(initialOptions, deps ?? {})) {
```

Comment:
I thought we are going to use reference equality for comparing builtin objects like `popupRedirectResolver`s and `persistence`s. Using deepEqual feels unnecessary.

Reply:
I was mainly using deepEqual to avoid extra lines of code, but also, while 2 of the 3 options (`popupResolver` and `errorMap`) can be compared using a simple `===`, `persistence` can be an array, and on top of that, an array where order matters.